roles/preflight-checks/tasks/check_items.yml

```diff
@@ -69,4 +69,16 @@
 - include: v7k.yml
   when: v7k.enabled
   tags: ['precheck_v7k']
-  run_once: true
\ No newline at end of file
+  run_once: true
+
+- name:  Make sure that if fixed_key is defined then cinder encryption is also enabled
+  assert:
+    that: "cinder.encryption.enabled"
+  when: cinder.encryption.fixed_key is defined
+  run_once: true
+
+- name:  Make sure that if cinder encryption is enabled, then only barbican or fixed_key is used (XOR)
```

Comment:
How about we allow both barbican.enabled and fixed_key is defined?
We can choose barbican in our code if both provided.